crates/subspace-networking/src/node.rs

```diff
@@ -320,37 +288,22 @@ impl Node {
             .map_err(PublishError::Publish)
     }
 
-    // Sends the pieces-by-range request to the peer and awaits the result.
-    pub async fn send_pieces_by_range_request(
-        &self,
```

Comment:
I would keep fine-grained interface for the node and make `send_generic_request` private. This way it's easy to discover supported logic for the `Node`, on the other hand if we will configure different protocols for different node types (farmer, subspace-node) the single `send_generic_request` hides unsupported protocols.

Reply:
The idea as I mentioned before is to move all request/response handlers from `subspace-networking` into a separate crate. Networking shouldn't really know anything about specific protocols and this PR achieves that. We may add a trait that implements handy methods on `Node` if that is desirable, but arguably it'll not bring too much value (though will help discoverability of the available functionality).

Not having generic method means we need to have ever growing API of `Node` instead of it being extensible from the outside without any changes here.